### PR TITLE
Check duplicate positions

### LINKED ISSUE #888: SgkitSampleData allows duplicate positions
It's common to get duplicate positions in VCF files, and these can be reasonably read in to SGKit. I think we should fail with a sensible error if we detect duplicate positions on conversion to an SgkitSampleData file, though. 

Edit - should fail if neither duplicate site is masked out.

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #888

### CODE REVIEW

**Comment by hyanwong** [comment]
Should probably allow duplicate positions if they are masked

File: tsinfer/formats.py [hunk 1]
```diff
@@ -2309,6 +2309,11 @@ def __init__(self, path):
                     " sgkit dataset, indicating that all the genotypes are"
                     " unphased"
                 )
+        if np.unique(self.sites_position).shape[0] != self.sites_position.shape[0]:
```
Comment (jeromekelleher) [inline tsinfer/formats.py:2312]: There's no need for a unique here, that's much more expensive than necessary. Can just say the np.diff must be greater than 0

Comment (hyanwong) [reply tsinfer/formats.py]: Oh, good catch. Yes, your suggested check should be fine. I have just updated the PR

**Comment by hyanwong** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@mergifyio rebase

**Review by benjeffery — APPROVED** [review]

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**